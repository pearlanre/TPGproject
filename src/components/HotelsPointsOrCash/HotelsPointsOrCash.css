@import url(https://fonts.googleapis.com/css?family=Roboto:300);
.hotel-title{
    margin: 30px 15px 15px 15px;
}