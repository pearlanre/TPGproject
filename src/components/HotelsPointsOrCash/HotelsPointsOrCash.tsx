import React from 'react';

import { ILoyalty } from '../../interfaces/ILoyalty';
import { PointsOrCash } from '../../utilities/pointsOrCash/PointsOrCash';

interface Props {
  data: ILoyalty;
}

export const HotelsPointsOrCash: React.FunctionComponent<Props> = ({
  data,
}: Props) => {
  return (
    <div>
      <p>Select A Hotel Rewards Program</p>
      <PointsOrCash data={data} program="Hotel" />
    </div>
  );
};

export default HotelsPointsOrCash;
