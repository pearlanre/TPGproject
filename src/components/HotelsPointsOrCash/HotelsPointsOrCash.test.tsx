import React from 'react';
import { shallow, mount } from 'enzyme';
import { HotelsPointsOrCash } from './HotelsPointsOrCash';

describe('AirLinePointsOrCash', () => {
  const props = {
    data: {
      name: 'Private hotel',
      tpg_valuation: 2,
      type: 'hotel',
    },
    program: '',
  };

  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<HotelsPointsOrCash {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should render the PointsOrCash Component correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should render HotelPointToCash', () => {
    const hotel = 'Select A Hotel Rewards Program';
    expect(wrapper.find('p').text()).toEqual(hotel);
    expect(wrapper).toHaveLength(1);
  });
});
