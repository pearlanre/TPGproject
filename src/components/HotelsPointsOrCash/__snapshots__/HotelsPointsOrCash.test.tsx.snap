// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AirLinePointsOrCash should render the PointsOrCash Component correctly 1`] = `
<div>
  <p>
    Select A Hotel Rewards Program
  </p>
  <PointsOrCash
    data={
      Object {
        "name": "Private hotel",
        "tpg_valuation": 2,
        "type": "hotel",
      }
    }
    program="Hotel"
  />
</div>
`;
